### PR TITLE
DSA Assignment 4

### PR DESCRIPTION
**Pull request by meta-rahul-kumar2**
DSA Assignment 4

### CODE REVIEW

File: DSA4/EmployeeCollection/src/Service/EmployeeCollection.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	/**
+	 * Method to add employee in the list
+	 * @param employee : employee object
+	 * @throws Exception
+	 */
+	public void addEmployee(EmployeePOJO employee) throws Exception{
+		if (employee == null) {
+			throw new Exception("Null object can't be added.");
+		}
+		else if(isUniqueEmployee(employee.getEmpId())){
```
[the remark this answers is not in the document]
Comment (meta-rahul-kumar2) [reply DSA4/EmployeeCollection/src/Service/EmployeeCollection.java]: if i use SET than , data type for Set is to be Employee object , but we have to check the duplicacy  by the employee id which is a member variable of Employee class , so i use Map and check duplicacy by own

[the remark this answers is not in the document]
Comment (meta-rahul-kumar2) [reply DSA4/EmployeeCollection/src/Service/EmployeeCollection.java]: If i use Set than i need to create a other Set variable of integer type to remove duplicacy by employee id

[the remark this answers is not in the document]
Comment (meta-manish) [reply DSA4/EmployeeCollection/src/Service/EmployeeCollection.java]: Can you implement comparator vs comparable

[the remark this answers is not in the document]
Comment (meta-rahul-kumar2) [reply DSA4/EmployeeCollection/src/Service/EmployeeCollection.java]: Sir comparable interface implements natural sorting order whereas comparator interface implements customized sorting order ,as comparable is used for sorting based on employee id  and comparator is used for sorting based on employee name

[the remark this answers is not in the document]
Comment (meta-manish) [reply DSA4/EmployeeCollection/src/Service/EmployeeCollection.java]: I mean we can override equals methods and all
but yes we can use Map right?

**Pull request merged**